src/files-rename.py

```diff
@@ -0,0 +1,24 @@
+import ctypes
+import os
+import sys
+
+AT_FDCWD = -100
+
+RENAME_NOREPLACE = (1 << 0)
+# RENAME_EXCHANGE = (1 << 1)
+
+libc6 = ctypes.CDLL(None, use_errno=True)
+
+if not libc6.renameat2:
+    sys.stderr.write("renameat2 not available")
+    sys.exit(1)
+
+source = sys.argv[1]
+dest = sys.argv[2]
+# overwrite = sys.argv[3]
+
+if libc6.renameat2(AT_FDCWD, source.encode(), AT_FDCWD, dest.encode(), RENAME_NOREPLACE) == -1:
+    sys.stderr.write(os.strerror(ctypes.get_errno()))
```

Comment:
I feel like I'd like to see some sort of structured json output similar to the thing that fsinfo reports on errors...

At the least, though: we should have some way of uniquely identifying the EEXIST case, no?

Reply:
I was considering returning a different exit code, json feels a bit too overengineered. There are only two scenario's:
* All Good = 0
* General error = 1
* File exists = 2

Reply:
I keep thinking that I Want to implement some kind of a strerror mechanism in javascript...

Reply:
I agree, not sure how I get errno.EEXISTS converted to str(). I feel this warrants a bigger discussion and some more thinking outside this PR. I do like the idea!